### ISSUE TITLE
[OneDrive] Add .md fallback logic

### ISSUE THREAD

**Issue by kptdobe** [adobe/helix-pages#125]
Once `fstab` is set into the project, a request to `/index.html` would try to fetch `/index.docx` in the corresponding OneDrive repo. If the file does not exist, request just fails. 
For https://github.com/davidnuescheler/theblog/issues/1 and certainly for other cases, we would need to manage md files instead of docs files in certain cases (like importing legacy read-only content). 
The request processing should then try:
- `index.docx` in OneDrive
- if no docx file, try `index.md` in OneDrive
- if no md file, try `index.md` in the git repository
- (if no md file in git repo, try `index.html` in the git repository) - no required for now

I did not test with Google Drive but I assume the same might apply.

cc @tripodsan

**Comment by trieloff** [comment]
I agree in principle, but not in detail: all candidate resources should be fetched at the same time, and the fallback logic should be evaluated after the first request has finished. Never wait for the first request to fail to make a second request.

**Comment by tripodsan** [comment]
I would add the fallback logic in word2md / docs2md and not in the pipeline. I think that requesting an explicit `.docx` in from the pipeline is wrong. we should just request the resource path, w/o the .docx extension.

**Comment by kptdobe** [comment]
We need this for theblog project.

**Comment by trieloff** [comment]
What about adobe/helix-dispatch#140? If we update helix publish to use v3, we should be good to go

**Comment by tripodsan** [comment]
> What about [adobe/helix-dispatch#140](https://github.com/adobe/helix-dispatch/pull/140)? If we update helix publish to use v3, we should be good to go

I think the ask here is that we want to be able to store `.md` files in onedrive, not a generic `default.md` as fallback.

**Comment by tripodsan** [comment]
> We need this for theblog project.
> 
> @tripodsan would you like to take care of this ? :)

ok, I can take a look.
